packages/automerge-repo/src/DocCollection.ts

```diff
@@ -44,28 +43,25 @@ export class DocCollection extends EventEmitter<DocCollectionEvents> {
    * Creates a new document and returns a handle to it. The initial value of the document is
    * an empty object `{}`. Its documentId is generated by the system. we emit a `document` event
    * to advertise interest in the document.
+   *
+   * @param {Object} opts - options
+   * @param {DocumentId} [opts.documentId] - a unique identifier for the newly created document. if not specified, one will be generated randomly.
+   * @param {T & object} [opts.schema] - the initial value of this document.
+   *   This is useful if the document must have some fixed schema. The document
+   *   will be initialized so that, if another `Repo` calls `create` with the same
+   *   `schema` and `documentId`, updates to fields within the `schema` can be
+   *   merged. See https://automerge.org/docs/cookbook/modeling-data/#setting-up-an-initial-document-structure.
    */
-  create<T>(): DocHandle<T> {
-    // TODO:
-    // either
-    // - pass an initial value and do something like this to ensure that you get a valid initial value
-
-    // const myInitialValue = {
-    //   tasks: [],
-    //   filter: "all",
-    //
-    // const guaranteeInitialValue = (doc: any) => {
-    // if (!doc.tasks) doc.tasks = []
-    // if (!doc.filter) doc.filter = "all"
-
-    //   return { ...myInitialValue, ...doc }
-    // }
-
-    // or
-    // - pass a "reify" function that takes a `<any>` and returns `<T>`
-
-    const documentId = uuid() as DocumentId
-    const handle = this.#getHandle<T>(documentId, true) as DocHandle<T>
+  create<T>(opts?: {
+    schema?: T & object
```

Comment:
AFAICT `automerge` expects the root of a document to be map-like, but this expectation isn't encoded into the type system for it or `automerge-repo` (e.g. with `DocHandle<T extends object>`). However, knowing that `schema` is an `object` makes the deep copy below a bit nicer.